src/main/java/net/coderbot/iris/shaderpack/IdMap.java

```diff
@@ -70,10 +78,15 @@
 
 		// TODO: Properly override block render layers
 
-		if (blockPropertiesMap == null) {
-			// Fill in with default values...
+		// Fill in with default values...
+		if (blockPropertiesMap == null || blockPropertiesMap.isEmpty()) {
 			blockPropertiesMap = new Object2IntOpenHashMap<>();
-			LegacyIdMap.addLegacyValues(blockPropertiesMap);
+			LegacyIdMap.addLegacyBlockValues(blockPropertiesMap);
+		}
+
+		if (dimensionIdMap == null || dimensionIdMap.isEmpty()) {
```

Comment:
This is invalid - if a shader pack specifies an empty dimension.properties file, that means that all dimensions map to the root directory. Only the absence of a dimension properties file should be interpreted as the shader pack using legacy mappings.